packages/beacon-node/src/api/impl/beacon/state/index.ts

```diff
@@ -254,8 +254,28 @@ export function getBeaconStateApi({
       }
 
       const epoch = filters.epoch ?? computeEpochAtSlot(state.slot);
+      const stateEpoch = computeEpochAtSlot(state.slot);
+      if (filters.slot !== undefined) {
+        const epochStartSlot = computeStartSlotAtEpoch(epoch);
+        const epochEndSlot = epochStartSlot + 32 - 1;
+
+        if (filters.slot < epochStartSlot || filters.slot > epochEndSlot) {
+          throw new ApiError(400, `Slot ${filters.slot} is not in epoch ${epoch}`);
+        }
+      }
+      if (Math.abs(epoch - stateEpoch) > 1) {
+        throw new ApiError(400, `Invalid epoch value: epoch ${epoch} is out of range`);
+      }
       const startSlot = computeStartSlotAtEpoch(epoch);
-      const decisionRoot = stateCached.epochCtx.getShufflingDecisionRoot(epoch);
+      let decisionRoot: string;
+      try {
+        decisionRoot = stateCached.epochCtx.getShufflingDecisionRoot(epoch);
+      } catch (error) {
+        if (error instanceof Error && error.message?.includes("EPOCH_CONTEXT_ERROR_DECISION_ROOT_EPOCH_OUT_OF_RANGE")) {
+          throw new ApiError(400, `Invalid epoch value: epoch ${epoch} is out of range`);
+        }
```

Comment:
as noted in https://github.com/ChainSafe/lodestar/pull/7941#discussion_r2215850764 I don't see why this is necessary, we checks the epoch above to make sure it's within one epoch of state epoch